geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/LiveFeatureCacheBenchmarkTest.scala

```diff
@@ -0,0 +1,395 @@
+    builder.set("Who", getName)
+    builder.set("What", Random.nextInt(10))
+    builder.set("When", randDate)
+    builder.set("Where", getPoint)
+    if (Random.nextBoolean()) {
+      builder.set("Why", string)
+    }
+    builder.buildFeature(i.toString)
+  }
+
+  def mean(values: Seq[Long]): Double = {
+    values.sum.toDouble / values.length.toDouble
+  }
+
+  def sd(values: Seq[Long]): Double = {
+    val mn = mean(values)
+    math.sqrt(
+      values.map(x => math.pow(x.toDouble - mn, 2.0)).sum /
+        (values.length - 1).toDouble)
+  }
+
+  def fd(value: Double): String = {
+    "%.1f".format(value)
+  }
+
+  def runQueries[T](n: Int, genIter: T => Long, filters: Seq[T]) = {
+    println(Seq(
+      "c_max",
+      "c_min",
+      "t_max",
+      "t_mean",
+      "t_sd",
+      "t_min",
+      "filter"
+    ).mkString("\t"))
+    for (f <- filters) {
+      val timeRes = (1 to n).map(i => time(genIter(f)))
+      val counts = timeRes.map(_._1)
+      val times = timeRes.map(_._2)
+      println(Seq(
+        counts.max,
+        counts.min,
+        times.max,
+        fd(mean(times)),
+        fd(sd(times)),
+        times.min,
+        f.toString
+      ).mkString("\t"))
+    }
+  }
+
+  def runQueriesMultiple[T](n: Int,
+                            labels: Seq[String],
+                            genIter: Seq[T => Long],
+                            filters: Seq[T]) = {
+    val sep = "\t"
+    val header = (for (l <- labels; c <- Seq("count", "tmax", "tmean", "tsd", "tmin")) yield l + "_" + c)
+    print(header.mkString(sep))
+    println(sep + "filter")
+
+    for (f <- filters) {
+      val row = genIter.flatMap {
+        g => {
+          val timeRes = (1 to n).map(i => time(g(f)))
+          val counts = timeRes.map(_._1)
+          val times = timeRes.map(_._2)
+          Seq(
+            counts.max,
+            times.max,
+            fd(mean(times)),
+            fd(sd(times)),
+            times.min)
+        }
+      }
+      print(row.mkString(sep))
+      println(sep + f)
+    }
+  }
+
+  def runQueriesMultipleRaw[T](n: Int,
+                               labels: Seq[String],
+                               genIter: Seq[T => Long],
+                               filters: Seq[T]) = {
+    val sep = ","
+    val header = for (i <- 1 to filters.size ; l <- labels) yield s"f$i.$l"
+    println(header.mkString(sep))
+
+    for (i <- 1 to n) {
+      val row = for (f <- filters; g <- genIter) yield {
+        val (counts, t) = time(g(f))
+        t
+      }
+      println(row.toList.mkString(sep))
+    }
+  }
+
+  def time[A](a: => A) = {
+    val now = System.currentTimeMillis()
+    val result = a
+    //println("%f seconds".format( (System.nanoTime - now) / 1000000000.0 ))
+
+    (result, System.currentTimeMillis() - now)
+  }
+
+  def timeUnit[Unit](a: => Unit) = {
+    val now = System.currentTimeMillis()
+    a
+    System.currentTimeMillis() - now
+  }
+
+  val ab = ECQL.toFilter("Who IN('Addams', 'Bierce')")
```

Comment:
Did all my quickly-written-poorly-named filters get used?  Let's remove the unused ones.